deploy/test-environments/fleet_api/src/agents_enrolled.py

```diff
@@ -0,0 +1,81 @@
+"""
+Wait for agents to be enrolled for a given policies
+If the expected number of agents is not enrolled within the timeout, the test will fail
+"""
+import sys
+import time
+from api.agent_policy_api import get_agents
+import configuration_fleet as cnfg
+from state_file_manager import state_manager
+from loguru import logger
+
+TIMEOUT = 600
+
+
+def get_expected_agents_mapping() -> dict:
+    """
+    Returns:
+        map: The name of the policy and the number of agents expected to be enrolled
+    """
+
+    logger.info("Loading agent policies state file")
+    policies_map = {}
+    for policy in state_manager.get_policies():
+        policies_map[policy.agnt_policy_id] = policy.expected_agents
+    return policies_map
+
+
+def get_actual_agents_mapping() -> dict:
+    """
+    Returns:
+        map: The name of the policy and the number of agents enrolled
+    """
+
+    agents = get_agents(cfg=cnfg.elk_config)
+    policies_map = {}
+    for agent in agents:
+        if agent.policy_id not in policies_map:
+            policies_map[agent.policy_id] = 0
+        policies_map[agent.policy_id] += 1
+    return policies_map
+
+
+def verify_agents_enrolled() -> bool:
+    """
+    Verify that the expected number of agents are enrolled
+    """
+    expected = get_expected_agents_mapping()
+    actual = get_actual_agents_mapping()
+    result = True
+    for policy_id, expected_count in expected.items():
+        if policy_id not in actual:
+            result = False
+            logger.info(f"Policy {policy_id} not found in the actual agents mapping")
+            continue
+        if actual[policy_id] != expected_count:
```

Comment:
I think `else if` would make more sense here, no? since if the policy is missing, then we wouldn't need to check the enrolled agents in this policy (hence why you added the `continue`s). This way you can get rid of the `continue`s